lib/device.ex

```diff
@@ -0,0 +1,149 @@
+defmodule Onvif.Device do
+  alias Onvif.Device
+  alias Onvif.Discovery.Probe
+
+  @enforce_keys [:username, :password, :ip]
+
+  defstruct @enforce_keys ++
+              [
+                :auth_type,
+                :scopes,
+                :manufacturer,
+                :model,
+                :firmware_version,
+                :serial_number,
+                :hardware_id,
+                :datetime,
+                :ntp,
+                :media_service_path,
+                current_diff: 0,
```

Comment:
current_diff outside of datetime context isn't super clear. maybe `time_diff_from_system_secs`